tools/generate-doc-preview-index.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+#
+
+HEAD_COMMIT_SHA=$(git rev-parse HEAD)
+cd $(git rev-parse --show-toplevel)/target/doc
+
+{
+    echo '<!doctype html>'
+    echo '<html>'
+    echo '<head>'
+    echo '  <metadata charset="utf-8">'
+    echo "  <title>Doc preview for ${HEAD_COMMIT_SHA}</title>"
+    echo '</head>'
+    echo '<body>'
+    echo "  <h1>Doc preview for ${HEAD_COMMIT_SHA}</h1>"
+} > index.html
+
+doc_link () {
+    local package=$1
+    if [[ -d ${package} ]]; then
+        echo "    <li><a href='${package}/index.html'>${package}</a></li>" >> index.html
+    fi
+}
+
+echo '  <h3>AWS Services</h3><ul>' >> index.html
+for package in $(ls | grep aws_sdk_ | sort); do doc_link ${package}; done
+echo '  </ul>' >> index.html
+
+echo '  <h3>AWS Runtime Crates</h3><ul>' >> index.html
+for package in $(ls | grep aws_ | grep -v _smithy_ | sort); do doc_link ${package}; done
```

Comment:
This also includes the crates that started with `aws_sdk_`, which were already included in the previous group. You can use OR patterns instead of two invocations to `grep`. And the call to `sort` is unnecessary since `ls` output is sorted alphabetically and `grep` does not reorder matches.

So it can be simplified to:

```suggestion
for package in $(ls -d aws_* | grep -v '_smithy_\|_sdk_'); do doc_link ${package}; done
```

Reply:
Good catch!